server/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -440,55 +439,61 @@ private void startLookups(List<LookupBean> lookupBeanList)
-              } else {
-                LOG.error(
-                    "Failed to start lookup [%s]:[%s]",
-                    lookupBean.getName(),
-                    container
-                );
-                return null;
-              }
-            }
+        futures.add(
+            completionService.submit(
+                () -> {
+                  LookupExtractorFactoryContainer container = lookupBean.getContainer();
+                  LOG.info(
+                      "Starting lookup [%s]:[%s]",
+                      lookupBean.getName(),
+                      container
+                  );
+                  if (container.getLookupExtractorFactory().start()) {
+                    LOG.info(
+                        "Started lookup [%s]:[%s]",
+                        lookupBean.getName(),
+                        container
+                    );
+                    return new AbstractMap.SimpleImmutableEntry<>(lookupBean.getName(), container);
+                  } else {
+                    LOG.error(
+                        "Failed to start lookup [%s]:[%s]",
+                        lookupBean.getName(),
+                        container
+                    );
+                    return null;
+                  }
+                }
+            )
         );
       }
-      IntStream.range(0, lookupBeanList.size())
-               .forEach(i -> {
-                 try {
-                   final Future<AbstractMap.SimpleImmutableEntry<String, LookupExtractorFactoryContainer>> completedFuture = completionService
-                       .take();
-                   final AbstractMap.SimpleImmutableEntry<String, LookupExtractorFactoryContainer> lookupResult = completedFuture
-                       .get();
-                   if (lookupResult != null) {
-                     builder.put(lookupResult);
-                   }
-                 }
-                 catch (InterruptedException | ExecutionException e) {
-                   LOG.error(e, "Lookup loading interrupted.");
-                 }
-               });
+      for (int i = 0; i < futures.size(); i++) {
+        try {
+          final Future<AbstractMap.SimpleImmutableEntry<String, LookupExtractorFactoryContainer>> completedFuture = completionService
+              .take();
+          final AbstractMap.SimpleImmutableEntry<String, LookupExtractorFactoryContainer> lookupResult = completedFuture
+              .get();
+          if (lookupResult != null) {
+            builder.put(lookupResult);
+          }
+        }
+        catch (ExecutionException e) {
+          LOG.error(e, "Execution error during lookup loading.");
+        }
+      }
       stateRef.set(new LookupUpdateState(builder.build(), ImmutableList.of(), ImmutableList.of()));
     }
     catch (Exception e) {
       LOG.error(e, "Failed to finish lookup load process.");
+      for (Future future : futures) {
+        if (!future.isDone()) {
```

Comment:
isDone not needed -- see example in `ExecutorCompletionService`'s javadoc.

Reply:
Removed.